src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -1963,5 +1963,62 @@ public TSource FirstOrDefault(Func<TSource, bool> predicate, params TSource[] de
             Assert.Equal("String", typeInfo.Type.Name);
             Assert.NotEmpty(typeInfo.Type.GetMembers("Replace"));
         }
+
+        [Fact]
+        [WorkItem(557, "https://github.com/dotnet/roslyn/issues/557")]
+        public void TestLambdaWithError11()
+        {
+            var source =
+@"using System.Linq;
+
+public static class Program
+{
+    public static void Main()
+    {
+        var x = new {
+            X = """".Select(c => c.
+            Y = 0,
+        };
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlibAndSystemCore(source);
+            var tree = compilation.SyntaxTrees[0];
+            var sm = compilation.GetSemanticModel(tree);
+            var lambda = tree.GetCompilationUnitRoot().DescendantNodes().OfType<LambdaExpressionSyntax>().Single();
+            var eReference = lambda.Body.DescendantNodes().OfType<IdentifierNameSyntax>().First();
+            Assert.Equal("c", eReference.ToString());
+            var typeInfo = sm.GetTypeInfo(eReference);
+            Assert.Equal(TypeKind.Struct, typeInfo.Type.TypeKind);
+            Assert.Equal("Char", typeInfo.Type.Name);
+            Assert.NotEmpty(typeInfo.Type.GetMembers("IsHighSurrogate")); // check it is the char we know and love
+        }
+
+        [Fact]
+        [WorkItem(5498, "https://github.com/dotnet/roslyn/issues/5498")]
+        public void TestLambdaWithError12()
+        {
+            var source =
+@"using System.Linq;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var z = args.Select(a => a.
+        var foo = 
```

Comment:
This can be fixed though with a bit of lookahead (and i thought we already did this int he compiler in one place).  Basically, instead of blindly consuming 'var' as the name of ```a.```, we instead peek ahead.  If we see that we have something that would not be valid in the expression context we're in, we ask the following.

1. Is 'var' on the next line?
2. Is it followed by whitespace and then an identifier?

If both are true, then we should not consume it as a name.  Once we've done that, then parser error recovery comes into play.  I would *not* expect argument parsing to succeed here (absent a comma or other clear indicator that this is an arg, and especially given that it looks like the start of a statement), and i would expect statement parsing to say: "yup, this looks good".  We'd then bail out of argument parsing and go into statement parsing.

We did this with ParseVariableDeclarator for example.   This is how we make sure htat we parse the following properly:

```c#
                // C                    //<-- here
                // Console.WriteLine();
```